README.md

```diff
@@ -80,34 +77,56 @@ auto metric = raft::distance::DistanceType::L2SqrtExpanded;
 raft::distance::pairwise_distance(handle, input.view(), input.view(), output.view(), metric);
 ```
 
+### Python Example
+
+The `pylibraft` package contains a Python API for RAFT algorithms and primitives. The package is currently limited to pairwise distances, and we will continue adding more.
+
+The example below demonstrates computing the pairwise Euclidean distances between cupy arrays.
+```python
+import cupy as cp
+
+from pylibraft.distance import pairwise_distance
+
+n_samples = 5000
+n_features = 50
+
+input = cp.random.random_sample((n_samples, n_features), dtype=cp.float32)
+output = cp.empty((n_samples, n_samples), dtype=cp.float32)
+
+pairwise_distance(input, input, output, "euclidean")
```

Comment:
> Might be good to add that this is a low level API that is prioritizing efficiency and simplicity over being pythonic? 

This is a good point. I actually think it might make sense to expose a very lightweight CAI wrapper around the RMM device buffer like the one we're using in the tests and make the second argument optional so that the `RmmCAIWrapper` could be returned by default, similar to functions I've seen in Scipy. This gives a user the option of doing this:

```python
option1 = pairwise_distance(in1, in2, metric="euclidean")
```

or this

```python
pairwise_distance(in1, in2, out=option2, metric="euclidean")
```